src/dfmt/formatter.d

```diff
@@ -280,12 +296,18 @@ private:
         else if (currentIs(tok!"identifier"))
         {
             writeToken();
+            //dfmt off
             if (index < tokens.length && (currentIs(tok!"identifier")
-                    || ( index > 1 && ( isBasicType(peekBack(2).type) || peekBack2Is(tok!"identifier") ) &&
-                         currentIs(tok!("(")) && config.dfmt_space_before_function_parameters )
+                    || ( index > 1 && ( isBasicType(peekBack(2).type)
+                                        || peekBack2Is(tok!"identifier")
+                                        || peekBack2Is(tok!")")
+                                        || peekBack2Is(tok!"]") )
+                        && currentIs(tok!("("))
+                        && config.dfmt_space_before_function_parameters )
```

Comment:
`config.dfmt_space_before_function_parameters` is the expression that's probably the cheapest to evaluate and also the most likely to be false. It's probably a good idea to put that first in this chain of `&&`s.